pkg/executor/results.go

```diff
@@ -140,7 +143,11 @@ func WriteJobResults(resultsDir string, stdout, stderr io.Reader,
 	}
 
 	result.ExitCode = exitcode
-	return result, err
+	return result
+}
+
+func NewFailedResult(reason string) *models.RunCommandResult {
+	return &models.RunCommandResult{ErrorMsg: reason, ExitCode: -1}
 }
```

Comment:
Do not set the exit code if the underlying runtime does not provide one. There is no requirement to provide an exit code in all circumstances.